File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/CrearActivity.java
package com.micky.miprimeraaplicacion;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.micky.miprimeraaplicacion.backend.BaseDatos;

public class CrearActivity extends AppCompatActivity {
    private EditText editTextPaterno;
    private EditText editTextMaterno;
    private EditText editTextNombre;
    private BaseDatos baseDatos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_crear);

        // Inicializar la base de datos
        baseDatos = new BaseDatos(this, "nombre_de_la_base_de_datos", null, 1);


        // Referencias a los elementos de la interfaz
        editTextPaterno = findViewById(R.id.editTextPaterno);
        editTextMaterno = findViewById(R.id.editTextMaterno);
        editTextNombre = findViewById(R.id.editTextNombre);
        Button buttonCrear = findViewById(R.id.buttonCrear);
        Button buttonMostrar = findViewById(R.id.buttonMostrar);
        Button buttonVolver = findViewById(R.id.buttonVolver);

        // Configurar el botón para agregar una persona
        buttonCrear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                agregarPersona();
            }
        });

        // Configurar el botón para mostrar todas las personas
        buttonMostrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Aquí puedes implementar la lógica para mostrar todas las personas
                // Esto puede incluir abrir una nueva actividad o mostrar un diálogo con la lista de personas.
                Toast.makeText(CrearActivity.this, "Mostrar todas las personas", Toast.LENGTH_SHORT).show();
            }
        });

        // Configurar el botón para volver atrás
        buttonVolver.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish(); // Finalizar la actividad y volver atrás
            }
        });
    }

    // Método para agregar una persona a la base de datos
    private void agregarPersona() {
        String paterno = editTextPaterno.getText().toString().trim();
        String materno = editTextMaterno.getText().toString().trim();
        String nombre = editTextNombre.getText().toString().trim();

        if (!paterno.isEmpty() && !nombre.isEmpty()) {
            baseDatos.agregarPersona(paterno, materno, nombre);
            Toast.makeText(this, "Persona agregada correctamente", Toast.LENGTH_SHORT).show();
            // Limpiar los campos después de agregar la persona
            editTextPaterno.getText().clear();
            editTextMaterno.getText().clear();
            editTextNombre.getText().clear();
        } else {
            Toast.makeText(this, "Por favor ingresa al menos el paterno y nombre", Toast.LENGTH_SHORT).show();
        }
    }
}

File: PrimerKotlin/app/src/main/java/com/micky/primerkotlin/CrearActivity.kt
package com.micky.primerkotlin


import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.micky.primerkotlin.backend.BaseDatos
import com.micky.primerkotlin.backend.Lugares
import android.content.Intent


class CrearActivity : AppCompatActivity() {
    private lateinit var etNombre: EditText
    private lateinit var etDescripcion: EditText
    private lateinit var etLatitud: EditText
    private lateinit var etLongitud: EditText
    private lateinit var btnAgregar: Button
    private lateinit var btnMostrar: Button
    private lateinit var dbHandler: BaseDatos

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_crear)

        etNombre = findViewById(R.id.etNombre)
        etDescripcion = findViewById(R.id.etDescripcion)
        etLatitud = findViewById(R.id.etLatitud)
        etLongitud = findViewById(R.id.etLongitud)
        btnAgregar = findViewById(R.id.btnAgregar)
        btnMostrar = findViewById(R.id.btnMostrar)
        dbHandler = BaseDatos(this)

        btnAgregar.setOnClickListener {
            val nombre = etNombre.text.toString().trim()
            val descripcion = etDescripcion.text.toString().trim()
            val latitud = etLatitud.text.toString().trim().toFloatOrNull() ?: 0.0f
            val longitud = etLongitud.text.toString().trim().toFloatOrNull() ?: 0.0f

            if (nombre.isEmpty() || descripcion.isEmpty()) {
                Toast.makeText(this, "Por favor, ingresa todos los campos", Toast.LENGTH_SHORT).show()
            } else {
                val lugar = Lugares()
                lugar.nombre = nombre
                lugar.descripcion = descripcion
                lugar.latitud = latitud
                lugar.longitud = longitud

                val success = dbHandler.addLugar(lugar)
                if (success) {
                    Toast.makeText(this, "Lugar correctamente agregado", Toast.LENGTH_SHORT).show()
                    // Limpia los EditTexts después de agregar el lugar exitosamente
                    etNombre.text.clear()
                    etDescripcion.text.clear()
                    etLatitud.text.clear()
                    etLongitud.text.clear()
                } else {
                    Toast.makeText(this, "Error al agregar el lugar", Toast.LENGTH_SHORT).show()
                }
            }
        }

        btnMostrar.setOnClickListener {
            val intent = Intent(this, MostrarActivity::class.java)
            startActivity(intent)
        }
    }
}
